docs/poly-to-normal-variant/index.md

```diff
@@ -0,0 +1,345 @@
+
+::: tip
+
+The alternative forms of a normal variant are called *variant constructors*,
+while the forms of a polymorphic variant are called *variant tags* (and always
+start with `` ` ``).
+
+:::
+
+However, this results in a compilation error:
+
+<<< type-error.txt
+
+We'll come back to this error message later. For now, observe that the error
+disappears if we simply delete `>`:
+
+<<< Types.re#delete-refinement
+
+This fixes the syntax error so that we now have a correctly-defined variant
+type.
+
+## Refactor `discount` variable to use normal variant
+
+Refactor the `discount` derived variable inside `Promo.make` to use our new
+variant type by deleting all occurrences of `` ` `` in the switch expression:
+
+<<< Promo.re#discount-variant{3,6,9-10}
+
+Likewise, the render logic can be fixed by removing all the `` ` `` occurrences:
+
+<<< Promo.re#discount-render{2-4,13}
+
+## Type constructor and type variable
+
+The current definition of the `discount` type works fine, but it is more verbose
+than necessary. In particular, the `DiscountError` constructor's polymorphic
+tags don't need to be fully specified:
+
+<<< Types.re#delete-refinement{4-11}
+
+The variant tags inside `DiscountError` originate from our `Discount` module, so
+they shouldn't be needlessly constrained by the `discount` type defined inside
+the `Promo` module.
+
+Change the `discount` type to this:
+
+<<< Types.re#type-variable
+
+Now `discount` is a *type constructor* that takes a *type variable* named `'a`.
+A type constructor is not a fixed type---you can think of it as a function that
+takes a type and outputs a new type.
+
+Using a type variable does not sacrifice type safety, if you hover over the
+`discount` variable, you see that its type is now:
+
+```reason
+discount([> `MissingSandwichTypes
+          | `NeedMegaBurger
+          | `NeedOneBurger
+          | `NeedTwoBurgers ])
+```
+
+This is essentially the same type as before (other than the `>` symbol,
+explained in the next section), just written differently. By looking at the
+usage of the `discount` variable, OCaml can infer how to fill in the type
+variable and produce the fixed type shown above.
+
+The advantage of using a type variable for the definition of the
+`Promo.discount` type is that when you add, rename, or delete variant tags in
+`Discount`, you won't have to make corresponding edits to `Promo.discount`[^1].
+
+## `>` = "allow more than"
+
+In the type expression above, we once again see `>`, so let's see what it means.
+In polymorphic variant type expressions, it means "allow more than". In this
+case, it means that tags other than the four that are listed are allowed. For
+example, this type would be allowed:
+
+```reason{5-6}
+discount([| `MissingSandwichTypes
+          | `NeedMegaBurger
+          | `NeedOneBurger
+          | `NeedTwoBurgers
+          | `HoneyButter
+          | `KewpieMayo ])
+```
+
+When defining your own types, you will most often used *fixed* polymormorphic
```

Comment:
These are called "exact" in official documentation: https://ocaml.org/manual/5.1/types.html#sss:typexpr-polyvar.